src/wp-includes/blocks/legacy-widget.php

```diff
@@ -24,13 +24,9 @@ function render_block_core_legacy_widget( $attributes ) {
 		return '';
 	}
 
-	if ( method_exists( $wp_widget_factory, 'get_widget_object' ) ) {
-		$widget_object = $wp_widget_factory->get_widget_object( $attributes['idBase'] );
-	} else {
-		$widget_object = gutenberg_get_widget_object( $attributes['idBase'] );
-	}
 
-	if ( ! $widget_object ) {
+	$widget_key = $wp_widget_factory->get_widget_key( $attributes['idBase'] );
+	if ( ! $widget_key ) {
```

Comment:
```suggestion
	if ( '' === $widget_key ) {
```
Instead of an implicit loose comparison, I'd suggest testing specifically against the value of empty string. Why? 

- avoids type juggle in PHP
- more readable and understandable by saying: if an empty string, bail out
- future proof, especially giving how PHP is moving towards strict types

Why empty string? The `get_widget_key` method returns a string. If the key doesn't exist, it returns an empty string.